aiohttp/locks.py

```diff
@@ -0,0 +1,42 @@
+import asyncio
+import collections
+
+from .helpers import ensure_future
+
+
+class ErrorfulOneShotEvent:
```

Comment:
Changed to `EventResultOrError`, still long but at least more meaningful. Let me know if it suits better for you.